recipe-box-backend/src/main/java/com/endava/recipebox/controller/RecipeController.java

```diff
@@ -37,9 +37,14 @@ public ResponseEntity<List<RecipeDTO>> getAllPublicRecipesByName(@RequestParam S
         return ResponseEntity.ok(recipeService.getAllPublicRecipesByName(recipeName, mealType));
     }
 
+    @GetMapping("/private")
+    public ResponseEntity<List<RecipeDTO>> getAllPrivateRecipesByUserId(@RequestParam Long userId, @RequestParam(required = false)MealType mealType) {
+        return ResponseEntity.ok(recipeService.getAllPrivateRecipesByUserId(userId, mealType));
+    }
+
     @GetMapping("/search/private")
-    public ResponseEntity<List<RecipeDTO>> getAllPrivateRecipesByUserId(@RequestParam Long userId) {
-        return ResponseEntity.ok(recipeService.getAllPrivateRecipesByUserId(userId));
+    public ResponseEntity<List<RecipeDTO>> getAllPrivateRecipesByUserIdAndName(@RequestParam Long userId, @RequestParam String recipeName, @RequestParam(required = false)MealType mealType) {
```

Comment:
Approved with the mention that it would have been better to just add the recipeName as a request parameter inside the  "/private" endpoint and add the filter inside RecipeService's `getAllPrivateRecipesByUserId` method, since the only difference between the two service methods used in the endpoints would be an additional filter